modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/api.mustache

```diff
@@ -0,0 +1,139 @@
+{{>licenseInfo}}
+@file:Suppress("UnusedImport")
+
+package {{apiPackage}}
+
+{{#imports}}import {{import}}
+{{/imports}}
+
+import {{packageName}}.infrastructure.*
+import io.ktor.client.HttpClient
+import io.ktor.client.request.forms.formData
+import io.ktor.client.engine.HttpClientEngine
+import io.ktor.client.features.json.serializer.KotlinxSerializer
+import io.ktor.client.request.forms.FormPart
+import io.ktor.client.utils.EmptyContent
+import kotlinx.serialization.json.Json
+import io.ktor.http.ParametersBuilder
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+{{#operations}}
+@Suppress(
+    "unused",
+    "RemoveRedundantQualifierName",
+    "RemoveExplicitTypeArguments",
+    "RemoveRedundantCallsOfConversionMethods",
+    "DuplicatedCode",
+    "UnnecessaryVariable",
+)
+public class {{classname}} : ApiClientBase {
```

Comment:
It would be nice to have a conditional generation of an `open` modifier:
`public {{#openApiClass}}open {{/openApiClass}}class {{classname}} : ApiClientBase {`

**Motivation:** In our apps for every real `XyzApi` we create a `MockXyzApi` with mock responses and allow testers to switch the Api used at runtime. Its very useful either for development, testing or presentation purposes.

Reply:
Yeah probably, I'll write that on the list of stuff that's still needed here

Reply:
This should only be needed for the mocking though and might be an option to implement platform specific async, I'd think, because most other use cases should be fine with extension functions

Reply:
Just to be clear, by mocking the apis I mean to create a subclass that overrides all the api functions. We are using these `MockXyzApi`s for testing of mobile apps running on real phone - mainly in app testing builds.
Although, sometimes we use these mocks also in production builds when users are allowed to try the app in some kind of a "demo" mode.

I've just realized that not only the api class but also the functions would need to be conditionally marked as `open`.

Reply:
Yeah but the models and helper classes should be fine. Also don't use this yet, it will probably have breaking changes, especially regarding the return types and maybe some other stuff, there's also no real android build currently, but JVM should work fine

Reply:
Another option is to add an annotation like `@GeneratedApi` to the base class, then users needing open classes can apply the `kotlin-allopen` compiler plugin with something like:

```kotlin
allOpen {
    annotation("com.mypackage.infrastructure.GeneratedApi")
}
```